src/azure-cli/azure/cli/command_modules/storage/azcopy/util.py

```diff
@@ -23,35 +23,52 @@
 
 STORAGE_RESOURCE_ENDPOINT = "https://storage.azure.com"
 SERVICES = {'blob', 'file'}
-AZCOPY_VERSION = '10.1.0'
+AZCOPY_VERSION = '10.3.3'
 
 
 class AzCopy(object):
     def __init__(self, creds=None):
         self.system = platform.system()
         install_location = _get_default_install_location()
-        if not os.path.isfile(install_location):
-            install_dir = os.path.dirname(install_location)
-            if not os.path.exists(install_dir):
-                os.makedirs(install_dir)
-            base_url = 'https://azcopyvnext.azureedge.net/release20190423/azcopy_{}_amd64_10.1.0.{}'
-            if self.system == 'Windows':
-                file_url = base_url.format('windows', 'zip')
-            elif self.system == 'Linux':
-                file_url = base_url.format('linux', 'tar.gz')
-            elif self.system == 'Darwin':
-                file_url = base_url.format('darwin', 'zip')
-            else:
-                raise CLIError('Azcopy ({}) does not exist.'.format(self.system))
-            try:
-                _urlretrieve(file_url, install_location)
-                os.chmod(install_location,
-                         os.stat(install_location).st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
-            except IOError as err:
-                raise CLIError('Connection error while attempting to download azcopy ({})'.format(err))
-
         self.executable = install_location
         self.creds = creds
+        if not os.path.isfile(install_location) or self.check_version() != AZCOPY_VERSION:
+            self.install_azcopy(install_location)
+
+    def install_azcopy(self, install_location):
+        install_dir = os.path.dirname(install_location)
+        if not os.path.exists(install_dir):
+            os.makedirs(install_dir)
+        base_url = 'https://azcopyvnext.azureedge.net/release20191212/azcopy_{}_{}_{}.{}'
+
+        if self.system == 'Windows':
+            if platform.machine().endswith('64'):
+                file_url = base_url.format('windows', 'amd64', AZCOPY_VERSION, 'zip')
+            else:
+                file_url = base_url.format('windows', '386', AZCOPY_VERSION, 'zip')
+        elif self.system == 'Linux':
+            file_url = base_url.format('linux', 'amd64', AZCOPY_VERSION, 'tar.gz')
+        elif self.system == 'Darwin':
+            file_url = base_url.format('darwin', 'amd64', AZCOPY_VERSION, 'zip')
+        else:
+            raise CLIError('Azcopy ({}) does not exist.'.format(self.system))
+        try:
+            _urlretrieve(file_url, install_location)
+            os.chmod(install_location,
+                     os.stat(install_location).st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
+        except IOError as err:
+            raise CLIError('Connection error while attempting to download azcopy ({})'.format(err))
+
+    def check_version(self):
```

Comment:
is there any way to get az version from memory instead of start a new process? the output message may change which has potential to break the result